### ISSUE TITLE
RSBtankAtlasCentaur is missing avionics again.

### ISSUE THREAD

**Issue by lamont-granquist** [KSP-RO/RP-0#1905]
This MM patch is supposed to add to the avionics limit for the boosters:

https://github.com/KSP-RO/RP-0/blob/864d4b37bb6db419eb3068e98120acbe8e6ded0b/GameData/RP-0/Avionics.cfg#L589-L593

That is getting applied but isn't working:

```
[LOG 11:38:12.240] Applying update RP-0/Avionics/@PART[RSBtankAtlasCentaur]:FOR[RP-0] to RealScaleBoosters/Parts/Atlas/RSBtankAtlasCentaur.cfg/PART[RSBtankAtlasCentaur]
```

I don't see it showing up when filtering by avionics or see the right avionics in the PAW, etc.

It does seem to work fine for the RSBtankAresIstage2 booster, which does show up when filtering for avionics, and in the PAW, etc.

I wonder if #1680 broke it, although that makes little sense...  (indeed reverting back to the prior MM patch doesn't change anything)

**Comment by lamont-granquist** [comment]
MechJeb also doesn't run on the RSBtankAtlasCentaur but does run on the RSBtankAresIstage2 in the VAB.  But I do see ModuleAvionics and ModuleCommand correctly in the parts in the saved craft file.

A diff between the two parts is here:

https://gist.github.com/lamont-granquist/24e5d07c4ef8e9ca6ce28e39f1681d23

I don't see anything that gives me any hints there, if there's state that makes a difference it isn't being persisted to the craft file (weirdly though MJ is hanging all kinds of state off of the booster even though MJ doesn't run if only the booster is there)

**Comment by lamont-granquist** [comment]
Diff based on the MM cache is here:

https://gist.github.com/lamont-granquist/8a3c9fa5e3f44c61c54cd8525dd0e148

I wonder if massLimit has to be an integer and not a float...

**Comment by siimav** [comment]
> I wonder if massLimit has to be an integer and not a float...

Definitely not.

**Comment by siimav** [comment]
Yes, the value is internally a float. At the same time I would assume that KSP cfg parser would know how to cast an int to a float.

**Comment by lamont-granquist** [comment]
Yep that wasn't it.  I do admit that I'm starting to hit the looking-in-the-refrigerator-to-find-my-keys phase of debugging this, but I don't see what the difference is between the part that works and the part that doesn't...
